Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -372,6 +372,9 @@ theorem reverse_reverse {u v : V} (p : G.Walk u v) : p.reverse.reverse = p := by
   | cons _ _ ih => simp [ih]
 #align simple_graph.walk.reverse_reverse SimpleGraph.Walk.reverse_reverse
 
+theorem reverse_surjective {u v : V} : Function.Surjective (reverse : G.Walk u v → _) :=
```

Comment:
I think Involutive might be the correct thing to have here, maybe Bijective / Surj / Inj could be given from that but imo Involutive is enough

Reply:
I'm facing an issue here: `Function.Involutive` expects the function to have the same domain type and range type, but for reverse, the domain is of type `G.Walk u v` while its range is of type `G.Walk v u`.

Reply:
Oh yeah, that's a fundamental issue. How about for now we settle for `reverse_surjective` for now? If you want you can also throw in `reverse_injective` and `reverse_bijective`.

There's a way to state that it's involutive, but what comes to mind involves sigma types and isn't very nice:
```
theorem Walk.reverse_involutive :
    Function.Involutive
      (Sigma.map Prod.swap (fun _ => Walk.reverse) : (p : V × V) × G.Walk p.1 p.2 → (p : V × V) × G.Walk p.1 p.2) := by
  intro ⟨u, v⟩
  simp [Sigma.map]
```

Reply:
Oh right, good point! Yeah don't do involutive